utils/azure/create-scale-set.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Shamelessly stolen from https://github.com/lsds/sgx-lkl
+
+# This script creates one of two VM scale set variants for use in Azure Pipelines.
+# After the scale set is created, a corresponding pool has to be created
+# in the Azure DevOps project that uses the scale set.
+# The two supported variants are "linux" and "windows"
+# The created resource group has the name '<vmss-name>-rg' where the VMSS name
+# must be specified as script argument.
+
+# Notes on OS and temporary disk:
+# The OS disk can either be ephemeral or persisted.
+# Ephemeral OS disks are backed by local non-cloud storage and can
+# at most be the size of the cached storage (e.g. DC4s_v2 has 86 GiB cache size).
+# See also https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ephemeral-os-disks.
+# Persisted OS disks can have arbitrary sizes but have higher latency
+# and make reset/reimaging operations of scale sets slower.
+# The temporary disk is always backed by local non-cloud storage and has
+# similar characteristics as an ephemeral OS disk. The size of the disk
+# is prescribed by the VM sku.
+
+# Notes on Azure Pipelines and its scale set agent feature:
+# The Azure Pipelines agent (including the job folder and home directory)
+# is running from the OS disk when using the scale set agent feature.
+# It is currently not possible to use the (often larger) temporary disk.
+# Ephemeral OS disks are currently not supported for DC v2 series VMs.
+# This means the best option currently is to use persisted OS disks for
+# the SGX scale set and ephemeral OS disks for the non-SGX scale set.
+# Once ephemeral OS disks are supported for DC v2 series VMs this should
+# be re-evaluated, keeping in mind that the maximum ephemeral OS disk size
+# for DC4s_v2 would be 86 GiB.
+
+set -e
+
+expected_args=4
+if [[ $# != $expected_args ]]; then
+    echo "Usage: $0 <subscription> <region> <vmss-name> linux|windows"
+    echo "Example: $0 7150ce20-6afe... eastus llvm-build linux"
+    exit 1
+fi
+
+subscription=$1
+region=$2
+os=$4
+vmss_name="verona-$3-$os"
+resource_group_name="$vmss_name-rg"
+
+# Common settings
+sku=Standard_D16s_v3
+ephemeral_os_disk=true
+os_disk_size_gb=200
+
+# Windows not supported yet
+if [ "$os" != "linux" ]; then
+  echo "Only Linux supported for now"
+  exit 1
+fi
+
+# az vm image list --publisher Canonical --offer UbuntuServer --sku 18.04-LTS --output table --all
+image="Canonical:UbuntuServer:18.04-LTS:latest"
```

Comment:
This is a gen1 image, which is generally fine, except with newer SKUs like DCv2 that tend to go gen2 only. For the gen2 image, just `s/18.04-LTS/18_04-lts-gen2`.

Reply:
Right, we can change it later, if needed. The beauty of having a script is that it's a lot easier to do that, rather than change some obscure documentation elsewhere. :)